scripts/tslint/noIncrementDecrementRule.ts

```diff
@@ -0,0 +1,36 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static POSTFIX_FAILURE_STRING = "Don't use '++' or '--' postfix operators outside statements, for loops, or element access expressions.";
+    public static PREFIX_FAILURE_STRING = "Don't use '++' or '--' prefix operators.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new IncrementDecrementWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class IncrementDecrementWalker extends Lint.RuleWalker {
+
+    visitPostfixUnaryExpression(node: ts.PostfixUnaryExpression) {
+        super.visitPostfixUnaryExpression(node);
+        if (node.operator === ts.SyntaxKind.PlusPlusToken || node.operator == ts.SyntaxKind.MinusMinusToken) {
+            this.visitIncrementDecrement(node);
+        }
+    }
+
+    visitPrefixUnaryExpression(node: ts.PrefixUnaryExpression) {
+        super.visitPrefixUnaryExpression(node);
+        if (node.operator === ts.SyntaxKind.PlusPlusToken || node.operator == ts.SyntaxKind.MinusMinusToken) {
+            this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.PREFIX_FAILURE_STRING));
+        }
+    }
+
+    visitIncrementDecrement(node: ts.UnaryExpression) {

```

Comment:
This is not an override and won't be called by the RuleWalker's walk of the tree, right? I don't know what the usual style for lint rules is, but in the past I found it less confusing to give these non-recursive checker methods a different prefix, something like `checkIncrementDecrement`.